lib/core/src/Cardano/Wallet/Primitive/AddressDerivation.hs

```diff
@@ -495,27 +499,60 @@ instance
 
 -- | Encrypt a 'Passphrase' into a format that is suitable for storing on disk
 encryptPassphrase
-    :: MonadRandom m => Passphrase purpose -> m (Hash purpose)
-encryptPassphrase (Passphrase bytes) = do
-    salt <- getRandomBytes @_ @ByteString 16
-    let params = Parameters
-            { iterCounts = 20000
-            , outputLength = 64
-            }
-    return $ Hash $ BA.convert $ mempty
-        <> BS.singleton (fromIntegral (BS.length salt))
-        <> salt
-        <> BA.convert @ByteString (fastPBKDF2_SHA512 params bytes salt)
+    :: MonadRandom m
+    => PassphraseScheme
+    -> Passphrase purpose
+    -> m (Hash purpose)
+encryptPassphrase scheme pwd@(Passphrase bytes) = case scheme of
+    EncryptWithPBKDF2 -> do
+        salt <- getRandomBytes @_ @ByteString 16
+        let params = Parameters
+                { iterCounts = 20000
+                , outputLength = 64
+                }
+        return $ Hash $ BA.convert $ mempty
+            <> BS.singleton (fromIntegral (BS.length salt))
+            <> salt
+            <> BA.convert @ByteString (fastPBKDF2_SHA512 params bytes salt)
+
+    EncryptWithScrypt ->  do
+        salt <- Scrypt.Salt <$> getRandomBytes @_ @ByteString 32
+        let msg = preparePassphrase pwd
+        return $ Hash $ Scrypt.getEncryptedPass $
+            Scrypt.encryptPass Scrypt.defaultParams salt msg
+
+-- | Manipulation done on legacy passphrases before getting encrypted.
+--
+-- Yes there's CBOR. Yes it's hashed before being encrypted. You don't actually
+-- pronounce the 'l' in salmon. A lot of things in the world don't make sense
+-- but we still live just fine.
+preparePassphrase
```

Comment:
probably we will need to export it in the module as we will need it in integration test

Reply:
We already expose `encryptPassphrase` and `checkPassphrases`, and that's all which is needed :)

Reply:
indeed!